File: packages/discovery-provider/src/models/grants/grant.py
from sqlalchemy import (
    Boolean,
    Column,
    DateTime,
    ForeignKey,
    Integer,
    String,
    Text,
    text,
)

from src.models.base import Base
from src.models.model_utils import RepresentableMixin


class Grant(Base, RepresentableMixin):
    __tablename__ = "grants"

    blockhash = Column(Text, nullable=False)
    blocknumber = Column(Integer, ForeignKey("blocks.number"), nullable=False)
    grantee_address = Column(String, primary_key=True, nullable=False)
    user_id = Column(Integer, primary_key=True, nullable=False)
    is_revoked = Column(Boolean, nullable=False, server_default=text("false"))
    is_approved = Column(Boolean, nullable=True)
    is_current = Column(Boolean, primary_key=True, nullable=False)
    created_at = Column(DateTime, nullable=False)
    updated_at = Column(DateTime, nullable=False)
    txhash = Column(
        String,
        primary_key=True,
        nullable=False,
        server_default=text("''::character varying"),
    )

    def get_attributes_dict(self):
        return {col.name: getattr(self, col.name) for col in self.__table__.columns}

File: packages/common/src/adapters/grant.ts
import { full, HashId } from '@audius/sdk'

import { Grant } from '~/models/Grant'

export const grantFromSDK = (input: full.Grant): Grant => {
  return {
    grantee_address: input.granteeAddress,
    user_id: HashId.parse(input.userId) ?? null,
    is_revoked: input.isRevoked,
    is_approved: input.isApproved,
    created_at: input.createdAt,
    updated_at: input.updatedAt
  }
}
